app/src/ui/add-repository/create-repository.tsx

```diff
@@ -119,7 +129,12 @@ export class CreateRepository extends React.Component<ICreateRepositoryProps, IC
 
     try {
       await this.ensureDirectory(fullPath)
+      this.setState({ ...this.state, isValidPath: true })
     } catch (e) {
+      if (e.code === 'EACCES' && e.errno === -13) {
+        return this.setState({ ...this.state, isValidPath: false })
```

Comment:
Minor thing and unrelated to this change -`Dialog.onSubmit` is the prop we assign `createRepository` to, and it's signature is `() => void`. But `createRepository` has an inferred signature of `() => Promise<void>` due to the places below where we are returning `postError`.

This usage should be fine for the sake of this change, but maybe we should let `Dialog.onSubmit` take a promise-returning function too.